packages/main/src/SwitchTemplateContext.js

```diff
@@ -0,0 +1,31 @@
+import SwitchType from "./types/SwitchType";
+
+class SwitchTemplateContext {
+	static calculate(state) {
+		const semantic = state.type === SwitchType.Semantic;
+		const textOn = semantic ? "" : state.textOn;
+		const textOff = semantic ? "" : state.textOff;
+		const mainClasses = SwitchTemplateContext.getMainClasses(state);
+
+		const context = {
+			ctr: state,
+			textOn,
+			textOff,
+			tabIndex: state.disabled ? undefined : "0",
+			classes: { main: mainClasses },
+		};
+
+		return context;
+	}
+
+	static getMainClasses(state) {
+		return {
+			"ui5-switch": true,
+			"ui5-switch--disabled": state.disabled,
```

Comment:
classes look good to me, but we need to discuss if we are staying with that BEM

Reply:
> classes look good to me, but we need to discuss if we are staying with that BEM

For sure we should discuss it, I liked it as well.